mathics/builtin/numbers/algebra.py

```diff
@@ -501,7 +501,13 @@ def apply(self, expr, evaluation):
         if sympy_expr is None:
             return expr
         # Now, try to simplify using sympy
-        sympy_result = sympy.simplify(sympy_expr)
+        try:
+            sympy_result = sympy.simplify(sympy_expr)
+        except Exception:
+            # This handles the exception reported in 214.
+            # However, it seems that the crash is due to a
+            # problem in the implentation of the class SympyExpression
+            return expr
```

Comment:
This specific case I believe it is SymPy side.

Reply:
I agree. We should get rid of this comment (and the misspelling in it) since the Python try says the same thing. And the fact that it was reported in an issue called 214 really is irrelevant. If this arcane information is needed, it is in fact stored in the git commit message.

Reply:
Sorry for not responding before, but I had a busy day. I just left the message in case someone else had the chance to discover the origin of the exception.

Reply:
the comment has gone.

Reply:
Ok, but I think in discussion we have come to feel more flexible solution would be to wrap this in a context manager method off of the evaluation. 

Using that here will mean that we won't have to touch this piece of code again. And this part of the code can be used for testing context manager features, even if this new context manager doesn't have lots of bells and whistles. 

Also, while it would be good to cover everything, it doesn't have to be done all at once.